xchange-deribit/src/main/java/org/knowm/xchange/deribit/v2/Deribit.java

```diff
@@ -0,0 +1,49 @@
+package org.knowm.xchange.deribit.v2;
+
+import java.io.IOException;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import org.knowm.xchange.deribit.v2.dto.marketdata.response.*;
+
+@Path("/api/v2/public")
+@Produces(MediaType.APPLICATION_JSON)
+public interface Deribit {
+
+  @GET
+  @Path("get_instruments")
+  DeribitInstrumentsResponse getInstruments(@QueryParam("currency") String currency)
```

Comment:
please create a DeribitException similar to https://github.com/knowm/XChange/blob/develop/xchange-bittrex/src/main/java/org/knowm/xchange/bittrex/dto/BittrexException.java
and add it to each throws declaration here

in your case the DeribitException will have a field

```
  @JsonProperty("error")
  private DeribitError error;
```

this way you will not have to do response parsing in the error adapter

your DeribitException should also implement the HttpStatusExceptionSupport interface as I see you need it in the error adapter

you can find more explanation on how its done here: https://github.com/mmazi/rescu/wiki/Exception-handling